### ISSUE TITLE
Private Channels not displaying on Dimensional Transceivers

### ISSUE THREAD

**Issue by Drethic** [SleepyTrousers/EnderIO#2367]
While using 2.3.0.373 I am not able to see ( on the available list ) or create new Private channels.  The dimensionalTransceiver.json file is in world/enderio and has all the expected channels listed.
I have 15 channels listed in the file and from above it shows all of them saving.
My UUID matches between login and the json file.

I see a lot of channels saved on server load:
```
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
[10:46:37] [pool-7-thread-1/INFO] [EnderIO]: ServerChannelRegister: Dimensional Transceiver data saved to /home/minecraft/./world/enderio/dimensionalTransceiver.json
```
Not sure what other information may be helpful.

**Comment by tterrag1098** [comment]
Could you screenshot the private channels GUI? What do you mean by "can't create" private channels?

**Comment by Drethic** [comment]
@tterrag1098 here are the images with some basic descriptions at each step:
http://imgur.com/a/OTXZ9

Here is what was added to world/enderio/dimensionalTransceivers after Image 1:
```
{
    "name": "Private Test",
    "type": 1
  },
```

**Comment by Drethic** [comment]
@tterrag1098 I tried updating my client and server to 2.3.0.378 and the inability to create private channels (process to replicate documented in the imgur images) or access existing private channels (also documented in the imgur images) is still non-functional.  Is there more information I can provide that may help narrow this down?

**Comment by DonnerVarg** [comment]
tl;dr: I *had* this very problem on a server. I tested in single player for this post and I **no longer** have this problem.

I seem to have been having the same issue, though I'm playing on a server and don't have immediate access to logs. Client and server are using version 2.3.0.375 beta.

Dimensional Transceivers using private channels as of 1 or 2 weeks ago are still using them. (I don't know what might have changed at that time. No mods were updated between when I started using the Dimensional Transceivers without a problem and now.)
I do not see these in the Available list, but I do see them in the Send or Receive lists in the GUI wherever they existed before.
If I try to create a private channel, it shows up in the available list without the lock icon beside the name.
There were no public channels prior to this that would have persisted. (I'm the first to get into these on the server and kept all my usage on private channels.)
This affects power and item tabs. I assume it affects all tabs, but I used no others prior to this problem beginning.
The server has gone through several scheduled restarts since I noticed the issue.

I tested in single player and discovered that I do not have the issue there, Though I did not have an existing single player world with dimensional anchors, the issue that I was unable to create new private channels or see existing private channels was not present.

After testing in single player I returned to the server world and do not have the problem. I can see the previously existing private channels and I can create new channels that are private.

I have screenshots that I'll compile into an album on imgur if that would help.

**Comment by DonnerVarg** [comment]
Had the problem again as soon as I had to start a new instance. Again fixed by opening a single player world. I didn't even have to do anything like opening the Dimensional Transceiver GUI.

**Comment by tterrag1098** [comment]
So from what I'm reading, this only occurs on servers, not in singleplayer?

**Comment by DonnerVarg** [comment]
I haven't had the issue in single player, but I haven't played single
player with dimensional transceivers extensively.

**Comment by Drethic** [comment]
@tterrag1098 I just tested 2.3.0.400 on a server and private channels still do not seem to be created.  This is a brand new world so I cannot see if the other private channels show up.

**Comment by tterrag1098** [comment]
I cannot reproduce this whatsoever. Tested on a local server.

**Comment by Drethic** [comment]
I downloaded my server and ran it locally, was able to reproduce the locked icon next to private channel not being there.
I removed all mods except EnderCore, Ender IO, and ForgeEssentials Server, still not seeing the lock indicating it's a private channel.

Here is what I had during the minimal test:
Forge Version: forge-1.7.10-10.13.4.1481-1.7.10-universal
Server Startup Script: java -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -Xmx2048M -Xms2048M -jar forge-1.7.10-10.13.4.1481-1.7.10-universal.jar
Brand new world
Mods: EnderCore-1.7.10-.0.1.12_alpha, EnderIO-1.7.10-2.3.0.400_beta, and forgeessentials-1.7.10-1.4.0-beta8-server
Configs: All default.  I deleted the config folder before starting the server to ensure no prior data would taint the test.
Here are images from the minimal test world: http://imgur.com/a/C4TUI

Are there any other data, tests, setups, etc, I can try to help you reproduce this?

**Comment by tterrag1098** [comment]
So, this only happens if you have no singleplayer worlds created on that client?

**Comment by tterrag1098** [comment]
Also, could you please try without FE? That is not needed for testing this, and it contains a lot of hacky stuff.

**Comment by Drethic** [comment]
Going to list out my whole process hoping it helps identify the steps I took to reproduce this.  And after doing this run of tests, I have to agree that not having a singleplayer world may be part of the problem.
* Using Technic Launcher, I deleted the Vanilla install of MC I had.
* Verified that the modpack was deleted out of .technic/modpack.
* Installed version 1.7.10 of Vanilla.
* Added in the forge-1.7.10-10.13.4.1481-1.7.10-universal jar as modpack.jar in the bin folder
* Added  EnderCore-1.7.10-.0.1.12_alpha, and EnderIO-1.7.10-2.3.0.400_beta to the mod folder
* Installed a fresh Forge server, added in the same two Ender* mods.
* Once in game I went creative, put down a creative capacitor bank and a dimensional receiver.
* Added a private channel, verifying the lock was selected.
* No lock was shown on the right hand side.
* Verified the usernamecache.json was not in the vanilla modpack folder.
* Added the backup usernamecache.json from last night and relaunched vanilla.
* Original Private channel still showing as Public, however, new Private channel had the lock as expected.

**Comment by tterrag1098** [comment]
So it's a username<->UUID issue. Interesting. Will investigate later.

**Comment by tterrag1098** [comment]
In testing, the above commit fixes the issue, however I'm unsure of the effects that change will have elsewhere. Time will tell, it's really hard to test such things by myself.

**Comment by Drethic** [comment]
Do you have a compiled list of features?  My friends and I can go through them all server side to test them out for you.

**Comment by tterrag1098** [comment]
@Drethic The most important thing I need to be tested on servers is the Dimensional Transceiver. Make sure there are no new issues from this change.

**Comment by Baughn** [comment]
One player on my server is currently seeing this issue, running 2.3.0.414.

Does that include that commit?

Of note, the player in question has changed his in-game name previously, and this problem appeared *after* upgrading to that release.

**Comment by DonnerVarg** [comment]
Have him open a new creative world and access a dimensional anchor (gui).
Then try again on the server. Please report back if the problem remains or
not after that. (For my own curiosity, as this worked around the issue for
me, having had no name changes.)

**Comment by Baughn** [comment]
That fixed it.

**Comment by tterrag1098** [comment]
Yes, the problem is that the GUI on the dim trans uses the clientside username->UUID map (the usernamecache.json in the root MC folder). It needs to be changed to either request the UUID from mojang, or use the server data.

Opening a creative world creates the usernamecache.json for your client, so then it works.
